lib/engine/game/g_18_co.rb

```diff
@@ -210,7 +210,6 @@ def operating_round(round_num)
         Round::Operating.new(self, [
         Step::Bankrupt,
         Step::G18CO::Takeover,
```

Comment:
For 18CO, a corporation can gain trains during Takeover, which could cause a DiscardTrain to be required. I have a concern that putting it after Route will allow the corporation to run an extra train for a round.

Reply:
Maybe I could change up the logic here, if a step gets passed over in this loop because `active` returned false, it could be marked as inactive? Then in my case, DiscardTrain would be inactive, so I could reject steps that were either `inactive` or have had `passed` called?

https://github.com/tobymao/18xx/blob/891ed4f699ab5f4bea0d950775c1ec9b29e4412c/lib/engine/round/base.rb#L156

Reply:
As long as the `inactive` flags are cleared with every action, that seems like it would work.

Reply:
My brain is exploding a bit from that 18CO playthrough so I don't even know if that made sense ... but yes, essentially cache the result of active? until the gamestate changes.

Reply:
My first attempt at this only breaks 8 out of 10,512 games, not too bad.

Reply:
Turns out those games are all the ones that I pinned for #2843, I just forgot to pin them in my local copy of the DB 😆

Reply:
Did another local verification, this is all looking good. Thanks @ryandriskel!